src/join-9.sol

```diff
@@ -103,24 +109,24 @@ contract GemJoin9 {
     function join(address usr, uint256 wad) external {
         require(gem.transferFrom(msg.sender, address(this), wad), "GemJoin9/failed-transfer");
 
-        _join(usr);
+        uint256 _wad = _join(usr);
 
-        emit Join(usr, wad);
+        emit Join(usr, _wad);
```

Comment:
yeah, I've included the change for discussion as this case is fairly unique, but I agree in principle to follow the join interface, and this was done in the original refresh PR, before audit.
I'll proceed to revert the change for now if there are no strong objections.
Regarding the `wad` vs `amt`, it usually depends on the decimal precision that in this case should be `18` and therefore a `wad`.
On the proposed change I'm not sure as current version is currently undergoing audit and this seems more like a design change, but if there is general consensus in going into this direction, to follow the join interface and avoid overloading functions, it could be considered. I would double check with @gbalabasquer on that though as there could be implications for the `dss-proxy-actions`.

Reply:
Reverted `CS-5.1` in 6fdd8ce

Reply:
Regarding `wad` vs `amt` - it is not only about decimals but about input amount vs slpped amount - see `join-7` (which is actually very similar to this join, just assumes the token has a readable implementation contract).

I do suggest to do/discuss my proposed change (for having one `join()` function), as it is not very a big change for the audit reponse anyhow. It will remove the current discrepancy of one `join` logging `amt` and one join logging `wad`, make the interface identical to other joins, remove the overloading and align almost completely with `join-7`.

Reply:
It seems that the rationale in having separate `join` it's also to return the `amt` in the proxy actions `lockGemDirect` and `lockGemDirectAndDraw` (see https://github.com/makerdao/dss-proxy-actions/pull/28).
The suggested change could break the `join` interface as a return value would be required in the case `amt = 0`.

Reply:
Even if we keep the function overloading so that we have one function that is the same as most of the gemJoins interface (no return value), there is still the discrepancy of event emissions.
I mean one Join() event is logging the interface value regardless of what was actually slipped, and one Join() event is emitting what was actually slipped. So anyone parsing these events might get things wrong.

We can solve that by emitting `Join(usr, 0)` at the first join function.
This is not elegant, but arguably less missleading.
Another option is haveing a different event for the first function (it's not standard anyway).

Can discuss that as well next week.

Reply:
Digging more in the original PR #7, It seems that in order to avoid the redundant `note`, it was referred to the `join(usr)` function and so emitting the slipped amount.
This could be translated here by emitting the event in the internal function but from Gonza's review, it seems he advised to emit the respective events in the overloaded functions, following the rationale of being consistent with previous `join` event behaviour (emitting input value for `join(usr,wad)` IIRC.